contributing/contribute-code.md

```diff
@@ -0,0 +1,209 @@
+    margin-top: 4px;
+  }
+}
+
+@media only screen and (min-width: 960px) {
+  .contribute-grid {
+    flex-wrap: nowrap;
+  }
+
+  .contribute-grid .column {
+    flex: 0 0 25%;
+  }
+
+}
+
+.contribute-grid .panel {
+  height: 100%;
+  margin: 0;
+}
+
+.contribute-grid .panel-body {
+  padding: 10px;
+}
+
+.contribute-grid h2 {
+  margin: 0 0 10px 0;
+  padding: 0;
+  display: flex;
+  align-items: flex-start;
+}
+
+.contribute-grid .number {
+  margin-right: 0.25em;
+  font-size: 1.5em;
+  line-height: 0.9;
+}
+</style>
+
+<div class="contribute-grid">
+  <div class="column">
+    <div class="panel panel-default">
+      <div class="panel-body">
+        <h2><span class="number">1</span><a href="#consensus">JIRA Ticket: <br /> Get Consensus</a></h2>
+        <p>Agree on importance, relevance, scope of the ticket, discuss the implementation approach and find a committer willing to review and merge the change.</p>
+        <p><b>Only committers can assign a Jira ticket.</b></p>
+      </div>
+    </div>
+  </div>
+  <div class="column">
+    <div class="panel panel-default">
+      <div class="panel-body">
+        <h2><span class="number">2</span><a href="#implement">Implement</a></h2>
+        <p>Implement the change according to the <a href="">Code Style and Quality Guide</a> and the approach agreed upon in the JIRA ticket.</p>
+      </div>
+    </div>
+  </div>
+  <div class="column">
+    <div class="panel panel-default">
+      <div class="panel-body">
+        <h2><span class="number">3</span><a href="#review">Review</a></h2>
+        <p>Open a pull request and work with the reviewer.</p>
+        <p><b>Pull requests belonging to unassigned Jira tickets will not be reviewed or merged by the community.</b></p>
+      </div>
+    </div>
+  </div>
+  <div class="column">
+    <div class="panel panel-default">
+      <div class="panel-body">
+        <h2><span class="number">4</span><a href="#merge">Merge</a></h2>
+        <p>A committer of Flink checks if the contribution fulfills the requirements and merges the code to the codebase.</p>
+      </div>
+    </div>
+  </div>
+</div>
+
+<div class="row">
+  <div class="col-sm-12">
+    <div class="panel panel-default">
+      <div class="panel-body">
+        Note: <i>trivial</i> hot fixes such as typos or syntax errors can be opened as a <code>[hotfix]</code> pull request, without a JIRA ticket.
+      </div>
+    </div>
+  </div>
+</div>
+
+
+<div class="alert alert-warning" role="alert">
+    <b>Note:</b> The code contribution process has changed recently (May 2019). The community <a href="https://lists.apache.org/thread.html/1e2b85d0095331606ad0411ca028f061382af08138776146589914f8@%3Cdev.flink.apache.org%3E">decided</a> to shift the "backpressure" from pull requests to Jira, by requiring contributors to get consensus (indicated by being assigned to the ticket) before opening a pull request.
+</div>
+
+
+<a name="consensus"></a>
+
+### 1. Jira Ticket: Get Consensus
```

Comment:
I think, it would help new contributors to describe, who is responsible for starting the consensus discussion. Is the contributor expected to find a committer? If so, how to do this? Or can the contributor expect that the ticket will get the attention of a committer automatically?

Reply:
Good point! I'll update

Reply:
Good point!